fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ChildPropertyDeriver.java

```diff
@@ -349,11 +359,81 @@ private void tryColocate(HashDistributionSpec leftShuffleDistribution,
             PhysicalPropertySet leftLocalProperty = createPropertySetByDistribution(
                     createLocalByByHashColumns(leftScanDistribution.getShuffleColumns()));
 
+            // Prefer colocated join than replicated join,
+            // Remove replicated join properties
+            if (canDoReplicated) {
+                outputInputProps.remove(outputInputProps.size() - 1);
+            }
             outputInputProps.add(new Pair<>(PhysicalPropertySet.EMPTY,
                     Lists.newArrayList(leftLocalProperty, rightLocalProperty)));
+            return true;
+        }
+    }
+
+    private void tryReplicatedCrossJoin(ExpressionContext context) {
+        if (!ConnectContext.get().getSessionVariable().isEnableReplicationJoin()) {
+            return;
+        }
+
+        GroupExpression rightChild = context.getChildGroupExpression(1);
+        List<LogicalOlapScanOperator> scanLists = Lists.newArrayList();
+        Utils.extractOlapScanOperator(rightChild, scanLists);
+
+        if (scanLists.size() != 1) {
+            return;
+        }
+
+        LogicalOlapScanOperator rightScan = scanLists.get(0);
+        if (rightScan.canDoReplicatedJoin()) {
+            HashDistributionSpec rightScanDistribution = rightScan.getDistributionSpec();
+            PhysicalPropertySet rightLocalProperty = createPropertySetByDistribution(
+                    createLocalByByHashColumns(rightScanDistribution.getShuffleColumns()));
+            outputInputProps.add(new Pair<>(PhysicalPropertySet.EMPTY,
+                    Lists.newArrayList(PhysicalPropertySet.EMPTY, rightLocalProperty)));
```

Comment:
If the empty attribute is required on the left, can ensure that the leftmost part of the planFragment is the Scan node？

Reply:
Fixed